Composite/Plugins/Functions/WidgetFunctionProviders/StandardWidgetFunctionProvider/DataReference/HomePageSelectorWidgetFunction.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections;
+using System.Xml.Linq;
+using Composite.Core.WebClient.Renderings.Page;
+using Composite.Data;
+using Composite.Data.Types;
+using Composite.Functions;
+using Composite.Plugins.Functions.WidgetFunctionProviders.StandardWidgetFunctionProvider.Foundation;
+
+namespace Composite.Plugins.Functions.WidgetFunctionProviders.StandardWidgetFunctionProvider.DataReference
+{
+    /// <summary>    
+    /// </summary>
+    /// <exclude />
+    [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+    public sealed class HomePageSelectorWidgetFunction : CompositeWidgetFunctionBase
+    {
+        /// <exclude />
+        public HomePageSelectorWidgetFunction(EntityTokenFactory entityTokenFactory)
+            : base(CompositeName, typeof(DataReference<IPage>), entityTokenFactory)
+        {
+        }
+
+        private const string CompositeName = CommonNamespace + ".DataReference" + ".HomePageSelector";
+
+        /// <exclude />
+        public override XElement GetWidgetMarkup(ParameterList parameters, string label, HelpDefinition helpDefinition, string bindingSourceName)
+        {
+            return StandardWidgetFunctions.BuildStaticCallPopulatedSelectorFormsMarkup(
+                parameters: parameters,
+                label: label,
+                helpDefinition: helpDefinition,
+                bindingSourceName: bindingSourceName,
+                optionsGeneratingStaticType: typeof(HomePageSelectorWidgetFunction),
+                optionsGeneratingStaticMethodName: nameof(GetHomePages),
+                optionsGeneratingStaticMethodParameterValue: null,
+                optionsObjectKeyPropertyName: "Key",
+                optionsObjectLabelPropertyName: "Label",
+                multiSelect: false,
+                compactMode: true,
+                required: true,
+                bindToString: false);
+        }
+
+        /// <summary>
+        /// To be called through reflection
+        /// </summary>
+        /// <exclude />
+        public static IEnumerable GetHomePages()
+        {
+            foreach (var element in PageStructureInfo.GetSiteMap())
+            {
+                yield return new
```

Comment:
Could be optimized by returning K`eyValuePair<Guid, string>` to avoid building an anonymous type. Makes it possible as well to use `nameof(KeyValuePair<string, string>.Key)` and .Label further up to avoid magic strings

Reply:
Thank you, Pauli. This was not initial, but a revert commit, to reset my tree selector functionality changes, so we have everything as before, but with bug fixes. Still, I agree with your optimization proposal